src/nodes.ts

```diff
@@ -1,30 +1,51 @@
 import { Iri, JsonLdObj } from 'jsonld/jsonld-spec';
 
-export interface Nodes {
-    all(): Iterable<JsonLdObj>;
-
-    add(node: JsonLdObj): Promise<void>;
+export interface Nodes extends AsyncIterable<JsonLdObj> {
+    count(): Promise<number>;
 
     get(id: Iri): Promise<JsonLdObj>;
 
+    set(article: JsonLdObj): Promise<void>;
+
+    delete(id: Iri): Promise<void>;
+
     has(id: Iri): Promise<boolean>;
 }
 
 export default class InMemoryNodes implements Nodes {
     private nodes: { [key: string]: JsonLdObj } = {};
 
-    all(): Iterable<JsonLdObj> {
-      return Object.values(this.nodes);
+    private i = 0;
+
+    [Symbol.asyncIterator](): AsyncIterator<JsonLdObj> {
+      const article = this.get(Object.keys(this.nodes)[this.i]);
+      this.i += 1;
+      return {
+        next(): Promise<IteratorResult<JsonLdObj>> {
+          return Promise.resolve({
+            done: false,
```

Comment:
I think you need to set `done` to true to avoid the memory issue. Maybe something like:
```
done: !article,
```
would work?